scripts/coverage/coverage_capture.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+# Copyright (C) 2024 Intel Corporation
+# Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+set -e
+
+[ "$1" != "" ] && OUTPUT_NAME="$1" || OUTPUT_NAME="total_coverage"
+
+set -x
+
+lcov --capture  --directory . \
+--exclude "/usr/*" \
+--exclude "*/build/*" \
+--exclude "*/benchmark/*" \
+--exclude "*/examples/*" \
+--exclude "*/test/*" \
+--exclude "*/src/critnib/*" \
+--exclude "*/src/ravl/*" \
+--exclude "*proxy_lib_new_delete.h" \
+--output-file $OUTPUT_NAME || \
+	( echo "RETRY after ERROR !!!:" && \
+	lcov --capture --directory . \
+	--exclude "/usr/*" \
+	--exclude "*/build/*" \
+	--exclude "*/benchmark/*" \
+	--exclude "*/examples/*" \
+	--exclude "*/test/*" \
+	--exclude "*/src/critnib/*" \
+	--exclude "*/src/ravl/*" \
+	--exclude "*proxy_lib_new_delete.h" \
+	--ignore-errors mismatch,unused,negative,corrupt \
+	--output-file $OUTPUT_NAME )
+
+# Most common UMF source code directory on most GH CI runners
+COMMON_UMF_DIR=/home/runner/work/unified-memory-framework/unified-memory-framework
```

Comment:
This is equivalent of `GITHUB_WORKSPACE`, I think.

Reply:
Yes, but it can be different on different runners.

Reply:
Does it even matter? The same runner is certainly used for running this capture script and uploading the result. Action that starts this capture script expects the output in a path specified with the first argument that is later used to upload an artifact. So in this script we don't want to change the output path, I think.

Reply:
This script is run on many different runners with different workspace directories. All those different workspace directories have to be replaced with the given common one.